containers/AddFly.js

```diff
@@ -1,25 +1,50 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
-import FlyForm from '../components/FlyForm'
+import FlyForm from '../components/FlyForm';
+import { connect } from "react-redux";
+import { addFly, clearFlyEntry } from '../actions';
+import { addFlyToAPI } from "../ApiCalls";
 
-export default function AddFly() {
-  return (
-    <View style={styles.container}>
-      <FlyForm />
+class AddFly extends Component {
+    constructor(props) {
+        super(props);
+    }
 
-      <View style={{flex: 1, flexDirection: 'row', justifyContent: 'center'}}>
-        <TouchableOpacity style={styles.button}
-          onPress = {() => {alert('Go to Home Page')}}>
-          <Text style={styles.button}>Cancel</Text>
-        </TouchableOpacity>
+  addFly = () => {
+    const currentFlyEntry = this.props.currentFlyEntry
+    addFlyToAPI(currentFlyEntry)
+      .then(data => {
+          this.props.addFly(data.data);
```

Comment:
Great job! So whenever a fly is successfully posted to the API via `addFlyToAPI`, we are returned that new fly's data object which we then store to state via the `addFly` action creator, correct?

Reply:
Yes. If it is a successful request, the response of the successful POST request returns an object of the Fly that is submitted.  So, that response is then added to the FliesData in the Redux store.

Reply:
I think we need to make a conditional where if there is a bad response, then the response shouldn't be added to the FliesData.  Let's keep that in mind when we make our future updates. I believe one way to take on that problem is by adding a disable feature to the button where the submit button is disabled until all fields are entered in the Add Fly Form.

Reply:
Sounds good - I think that's a great idea.